src/Components/CardboardListCallout/CardboardListCallout.types.ts

```diff
@@ -52,6 +52,7 @@ export interface ICardboardListCalloutProps<T = void> {
 
 type PrimaryActionProps = {
     primaryActionLabel: string;
-    onPrimaryActionClick: () => void;
+    onPrimaryActionClick: (searchTerm?: string) => void;
```

Comment:
can't we just always send hte value and the consumer can decide whether to use the value? after all, they pass in the handler for this click action.

Reply:
I think the primary action and search term are not tightly related, that is why I added isSearchTermExposedWithPrimaryActionClick prop to expose the search term in onPrimaryActionClick only when it is set. I don't think we should always send the search term when we click on primary action click (e.g. cases where the primary action is irrelevant like learn more).

Reply:
there won't be a search term in the case that it's Learn More though, cause there won't be a searchbox shown right? Also, the consumer would know they dont' care about the value and can simply ignore the argument. You get the flexibility without any extra complexity of more props and edge cases on when you should expect an argument value or not.